src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/BaseDiagnosticAndGeneratorItemSource.cs

```diff
@@ -103,6 +103,9 @@ private async ValueTask ProcessQueueAsync(CancellationToken cancellationToken)
             this.Workspace.WorkspaceChanged -= OnWorkspaceChanged;
 
             _cancellationTokenSource.Cancel();
+
+            // Note: mutating _items will be picked up automatically by clients who are bound to the collection.  We do
+            // not need to notify them through some other mechanism.
             _items.Clear();
```

Comment:
It's fine, and I can doc.  Basically the BulkCollection supports changing on any thread. But if it's not on the ui thread, it itself schedules work to the ui thread. However, this work isn't tracked, which breaks unit tests.  

So we explicitly transition to the ui thread ourselves so that we can properly teach the work through the async listener, and so tests can work properly.

Reply:
So, why not do the UI thread switch here too? Wouldn't it possibly be needed by tests?

Reply:
i can do that too i guess :)  we just don't need it yet (not testing it).  but 'll make the change for consistency :)

Reply:
dnoe.